src/globus_sdk/services/search/data.py

```diff
@@ -221,6 +222,51 @@ def add_sort(
         return self
 
 
+class SearchQueryV1(utils.PayloadWrapper):
+    """
+    A specialized dict which has helpers for creating and modifying a Search
+    Query document. Replaces the usage of ``SearchQuery``.
+
+    :param q: The query string. Required unless filters are used.
+    :param limit: A limit on the number of results returned in a single page
+    :param offset: An offset into the set of all results for the query
+    :param advanced: Whether to enable (``True``) or not to enable (``False``) advanced
+        parsing of query strings. The default of ``False`` is robust and guarantees that
+        the query will not error with "bad query string" errors
+    :param additional_fields: additional data to include in the query document
```

Comment:
I'm a little surprised that pylint is passing, given that we have some undocumented parameters. Maybe I need to look at its settings for `__init__`... Regardless, we need param doc for `filters`, etc. It should be pretty short and easy to add.

Reply:
pylint catches that? PyCharm notifies me about parameter mismatches in docstrings but I haven't spotted that in other linters.

Reply:
Well, it's _supposed to_, under our config:
```toml
[tool.pylint]
load-plugins = ["pylint.extensions.docparams"]
accept-no-param-doc = "false"

[tool.pylint."messages control"]
disable = [  # simplified -- we have other disabled rules
    "missing-raises-doc",
]
```

But I'm guessing that it doesn't understand that the class docstring applies to the init method. Maybe it's configurable and we can fix it.